src/_blog/rust-libp2p-2022.md

```diff
@@ -0,0 +1,97 @@
+
+Our current implementation of WebRTC enables [browsers to connect to rust-libp2p based servers](https://github.com/libp2p/specs/tree/master/webrtc#browser-to-public-server) without those servers needing to have signed TLS certificates.
+QUIC is a better libp2p transport than the combination of TCP+Noise+Yamux in almost every dimension as it provides faster connection establishment, better native multiplexing, native encryption, and higher hole punching success rates.
+.Along the way, given that QUIC already requires TLS, [rust-libp2p can now secure TCP connections with TLS as well](https://github.com/libp2p/rust-libp2p/pull/2945) (previously only Noise).
+
+### User Experience Improvements
+Along the way we tackled many smaller improvements, as a whole having a big impact on the user experience.
+To mention a couple: [naming consistency across crates](https://github.com/libp2p/rust-libp2p/issues/2217), [refactoing of the many `inject_*` into a single `enum` event handler in both `NetworkBehaviour` and `ConnectionHandler`](https://github.com/libp2p/rust-libp2p/issues/2832), [deprecation of event-based `PollParameters`](https://github.com/libp2p/rust-libp2p/pull/3153), and the [rework of our Rust feature flags](https://github.com/libp2p/rust-libp2p/pull/2918).
+Still remember the old days with the `NetworkBehaviourEventProcess` trait?
+[All gone](https://github.com/libp2p/rust-libp2p/pull/2784) in favor of the much simpler (generated) `OutEvent` mechanism.
+The `StreamMuxer` trait received numerous significant simplifications, [basically rewriting](https://github.com/libp2p/rust-libp2p/issues/2722) the trait as well as the trait implementation in yamux, mplex and now QUIC and WebRTC.
+
+### DoS Protection
+Defense against denial-of-service attacks is a cornerstone of a networking library, especially in the peer-to-peer space.
+rust-libp2p saw a lot of related improvements in 2022.
+We enforce various limits (e.g. on the [number of streams](https://github.com/libp2p/rust-libp2p/pull/2697) and bytes of a request) and [prioritize local work over new incoming work from a remote](https://github.com/libp2p/rust-libp2p/pull/2627) across the many layers.
+Up next is [a patch](https://github.com/libp2p/rust-libp2p/issues/2824) enabling `NetworkBehaviour` implementations to implement their own connection management strategies.
+
+### Metrics and Observability
+Understanding large systems is hard.
+Understanding distributed systems is even harder.
+We made understanding large libp2p networks a bit easier in 2022 introducing a metric crate for rust-libp2p exposing Prometheus metrics, e.g. the [time to establish a connection](https://github.com/libp2p/rust-libp2p/pull/3134) or the [protocols supported by peers](https://github.com/libp2p/rust-libp2p/pull/2734).
+
+### Housekeeping & Quality of Life Changes
```

Comment:
Bit undecided, but I think this fits better into _Meta - Improvements to the Project_ than _Technical Highlights_.